Assets/Source/Components/Enemy/EnemySpawnerComponent.cs

```diff
@@ -0,0 +1,104 @@
+﻿using UnityEngine;
+using Assets.Source.Components.Base;
+using Assets.Source.Constants;
+
+namespace Assets.Source.Components.Enemy
+{
+    public class EnemySpawnerComponent : ComponentBase
```

Comment:
This is fine for now, but i think we will have a level director object (the state machine thing i was talking about) actually handle spawning enemies, since they'll be limited and killing all enemies theoretically advances the level to the next phase

Reply:
actually, i think we could use this.  

The level director object can handle spawning important enemies that need to be killed to advance, but this could be useful for spawning an unlimited bunch of smaller enemies.

Reply:
Sure why not utilize both? Or have the level director call methods in this class where everything could be configured? Just depends on how coupled we want this and where the logic lives. This also makes it easier for at least testing if nothing else for now.